### ISSUE TITLE
ERR_UNSUPPORTED_DIR_IMPORT when using MDXContent or useMDXComponent

### ISSUE THREAD

**Issue by rossmeissl** [sdorra/content-collections#277]
![image](https://github.com/user-attachments/assets/ba34bc8d-1d85-49d0-9f68-cfe94382715f)

If I use `mdx-bundler`'s `getMDXComponent` mechanism directly, the problem goes away.

**Comment by sdorra** [comment]
Could you provide more details? Which package manager do you use? Is your application a module (`type=module` in package.json) or a commonjs app? Do you use the MDX hook or component in an RSC context? Can you show the code which uses the MDX hook or component?

**Comment by rossmeissl** [comment]
@sdorra I could have sworn I responded here a few minutes ago but now it's gone. Did you remove the comment?

**Comment by sdorra** [comment]
@rossmeissl No, I am looking at the ticket for the first time today

**Comment by rossmeissl** [comment]
@sdorra OK strange let me try again:

* npm
* Next.js 14.2.5
* There's no `type=module` in `package.json` but I also don't think it's fair to say it's a CommonJS application
* Here's a screenshot of the implementation against `mdx-bundler` which is almost the same as yours, but it works correctly. (I tried both the component and hook.)
<img width="875" alt="image" src="https://github.com/user-attachments/assets/35a05071-dfb9-4f30-aff3-616b54c1dbc0">

**Comment by sdorra** [comment]
I've tried to reproduce the issue:

https://github.com/sdorra/cc-issue-277

But it works as expected. Can you check if there are any differences between my sample and yours?

**Comment by rossmeissl** [comment]
My directory structure uses `src/pages` instead of `/app` . . . could that be it?

**Comment by rossmeissl** [comment]
Unfortunately no it's our monorepo but let me see if I can extract just enough to repro . . .

**Comment by rossmeissl** [comment]
@sdorra sent you an email. Let me know if I can help further!

**Comment by sdorra** [comment]
@rossmeissl i was able to reproduce the issue. I can even reproduce it with my repository. When I navigate to an article and then press refresh, I get the error. The error seems to occur only in the Pages Router. I will try to fix it.

**Comment by rossmeissl** [comment]
Thanks @sdorra !

**Comment by sdorra** [comment]
@rossmeissl the fix is not yet released. Have you tested the fix with the preview release? If not can you please test if it fixes your problem? You only have to install the packages as described in the following comment:

https://github.com/sdorra/content-collections/pull/291#issuecomment-2331779390

**Comment by rossmeissl** [comment]
Unfortunately I think it still doesn't work. I updated `@content-collections/mdx` in my `package.json` as follows:

```json
{
  // ...
  "devDependencies": {
    "@content-collections/core": "^0.6.2",
    "@content-collections/mdx": "https://pkg.pr.new/sdorra/content-collections/@content-collections/mdx@291",
    // ...
  }
  // ...
}
```

And I get:

<img width="980" alt="image" src="https://github.com/user-attachments/assets/97e3c26b-b795-475c-bfd5-a66c8b6678a8">

**Comment by rossmeissl** [comment]
Actually, hmm, this appears to have just emerged unrelated to this PR

**Comment by rossmeissl** [comment]
@sdorra shall I close this?
